extension/oauth2clientauthextension/extension_test.go

```diff
@@ -186,13 +228,6 @@ func TestOAuthClientSettingsCredsConfig(t *testing.T) {
 			assert.NoError(t, err)
 			assert.Equal(t, test.expectedClientConfig.ClientID, cfg.ClientID)
 			assert.Equal(t, test.expectedClientConfig.ClientSecret, cfg.ClientSecret)
-
-			// test tls settings
-			transport := rc.client.Transport.(*http.Transport)
```

Comment:
Why was this removed?

Reply:
No, I think it's fine, thank you for the explanation. My concern was whether the new code introduced changes in how we handle TLS, which wasn't apparent at first for me.

Reply:
Actually, can you revert this change, and send a new PR removing it? Just so that we isolate the changes, ensuring they are not related.

Reply:
I had it in a separate commit. I pushed a reversion commit. I'll send a new PR when we move forward with this one.